src/checkers/adapters/git.history.adapter.js

```diff
@@ -0,0 +1,20 @@
+const { flow } = require('lodash');
+
+function adapt(base, { readGitHistoryFn, onCheckResult }) {
+  return function check(repo, context, config) {
+    return flow(
+      readGitHistoryFn,
+      (history) => {
+        const historyResults = base(history, context, config);
+
+        if (onCheckResult && historyResults.length) {
```

Comment:
Is it necessary to double-check the `historyResults.length` in this and `onCheckResult` functions?

Reply:
for this implementation yes, see related draft PR for more details.